aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -200,23 +200,79 @@ void copy_to_cpu(Tensor& dst, const Tensor& src) {
   });
 }
 
+void copy_from_cpu_async_(Tensor& dst, const Tensor& src) {
+  AT_ASSERT(dst.is_contiguous());
+  AT_ASSERT(src.is_contiguous());
+
+  if (dst.numel() == 0) {
+    return;
+  }
+
+  DeviceGuard device_guard{dst.device()};
+  CUDAStream stream = getCurrentCUDAStream();
+
+  AT_DISPATCH_ALL_TYPES_AND_HALF(src.type(), "copy_from_cpu_async", [&]() {
+    AT_CUDA_CHECK(cudaMemcpyAsync(
+        dst.data<scalar_t>(),
+        src.data<scalar_t>(),
+        src.numel() * sizeof(scalar_t),
+        cudaMemcpyHostToDevice,
+        stream));
+    AT_CUDA_CHECK(THCCachingHostAllocator_recordEvent(
+        src.storage().data<scalar_t>(), stream.internals()));
```

Comment:
I have a patch coming down the pipe that will conflict with this. We'll see who lands first :)